Src/AutoFixture.xUnit.net/FrozenAttribute.cs

```diff
@@ -11,22 +13,68 @@ namespace Ploeh.AutoFixture.Xunit
     [AttributeUsage(AttributeTargets.Parameter, AllowMultiple = false)]
     public sealed class FrozenAttribute : CustomizeAttribute
     {
+        private readonly Matching by;
+        private readonly string targetName;
+        private IRequestSpecification matcher;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FrozenAttribute"/> class.
+        /// </summary>
+        /// <param name="by">
+        /// The <see cref="Matching"/> criteria used to determine
+        /// which requests will be satisfied by the frozen parameter value.
+        /// The default value is <see cref="F:Matching.ExactType"/>.
+        /// </param>
+        /// <param name="targetName">
+        /// The identifier used to determine which requests
+        /// for a class member will be satisfied by the frozen parameter value.
+        /// </param>
+        public FrozenAttribute(
+            Matching by = Matching.ExactType,
+            string targetName = null)
+        {
+            this.by = by;
+            this.targetName = targetName;
+        }
+
         /// <summary>
         /// Gets or sets the <see cref="Type"/> that the frozen parameter value
         /// should be mapped to in the <see cref="IFixture"/>.
         /// </summary>
+        [Obsolete("The ability to map a frozen object to a specific type is deprecated and will likely be removed in the future. If you wish to map a frozen type to its implemented interfaces, use [Frozen(By = Matching.ImplementedInterfaces)].")]

```

Comment:
This is a good and helpful message already. Should we also add a sentence about how to map to a base class, if that's the user's case?